scapy/contrib/cansocket_native.py

```diff
@@ -122,8 +147,13 @@ def send(self, x):
         # required by the underlying Linux SocketCAN frame format
         bs = raw(x)
         if not conf.contribs['CAN']['swap-bytes']:
-            bs = bs + b'\x00' * (CAN_MTU - len(bs))
-            bs = struct.pack("<I12s", *struct.unpack(">I12s", bs))
+            print(type(x))
+            if type(x) is CANFD:
```

Comment:
use ```isinstance(...)```

Reply:
ok.. but what if someone provides a new random class that should be sent as CANFD? 
Can't you make this decision here, based on the pkt length?

Reply:
you if someone uses a cansocket for non CAN/CANFD pakets which than have more than 8 bytes of data?